src/kernel-install/90-uki-copy.install

```diff
@@ -26,12 +26,12 @@ KERNEL_VERSION="${2:?}"
 ENTRY_DIR_ABS="$3"
 KERNEL_IMAGE="$4"
 
-[ "$KERNEL_INSTALL_LAYOUT" = "uki" ] || exit 0
+[ "$KERNEL_INSTALL_IMAGE_TYPE" = "uki" ] || [ -f "$KERNEL_INSTALL_STAGING_AREA/uki.efi" ] || exit 0
 
 ENTRY_TOKEN="$KERNEL_INSTALL_ENTRY_TOKEN"
 BOOT_ROOT="$KERNEL_INSTALL_BOOT_ROOT"
 
-UKI_DIR="$BOOT_ROOT/EFI/Linux"
+UKI_DIR="$(bootctl --print-esp-path)/EFI/Linux"
```

Comment:
`kernel-install` sets `KERNEL_INSTALL_BOOT_ROOT` (see `BOOT_ROOT_SEARCH` there). Also, it is configurable, so there might be cases where `KERNEL_INSTALL_BOOT_ROOT`, when set from the outside, and `bootctl --print-esp-path` might not agree.

Reply:
Yes.  That is the whole point.  `$KERNEL_INSTALL_BOOT_ROOT` might point to something which is not the ESP.  On a typical fedora install for example you have `BOOT=/boot` and `ESP=/boot/efi`.  The unified kernel needs to go to the ESP though.

Reply:
But the ESP can be mounted at `/boot` and [BLS](https://uapi-group.org/specifications/specs/boot_loader_specification/#the-boot-partition-placeholder). In the logic in `kernel-install` `BOOT_ROOT` will be either `/boot`, `/efi` or `/boot/efi` (although the latter is something that is not recommended), i.e. this will already be set to the correct value outside of the plugin.

My point above is still different, though: `KERNEL_INSTALL_BOOT_ROOT` can be set externally for the invocation of `kernel-install` via the `BOOT_ROOT` environment variable. It will break backward compat if that is ignored and something else is used in its place.

Reply:
Yeah, this change just makes this worse. Please just let `$BOOT_ROOT` do its job.